transaction/src/manifest/decompiler.rs

```diff
@@ -28,7 +32,10 @@ pub fn decompile(manifest: &TransactionManifest) -> Result<String, DecompileErro
                 buckets.insert(bucket_id, name.clone());
                 buf.push_str(&format!(
                     "TAKE_FROM_WORKTOP ResourceAddress(\"{}\") Bucket(\"{}\");\n",
-                    resource_address, name
+                    bech32_encoder
+                        .encode_resource_address(&resource_address)
+                        .unwrap(),
```

Comment:
Should return a DecompileError and similarly for other unwraps in this file.

Reply:
Yup, changed that. Will push the changed with the `address` macro once I'm done :)